From 18b2e91b915311086d85570c713a73b7f35e1a51 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Thu, 01 Feb 2024 14:07:00 +0200
Subject: [PATCH] [SW-173074] gaudi3: add delay for fifo pop

It is observed that after setting pop bit in
fifo control register it takes little time to
remove the entry from the fifo. This patch
adds 1 micro second delay after setting the
pop bit without this delay the entries in the
fifo does not get reduced when pop is executed.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I3315826ee7e1ae36142a87a62448e0151a261089
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 771b825..97bae5d 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -223,6 +223,7 @@
 		preboot_cfg_dma_mmu_bypass(false);
 }
 
+#define HL_BTM_FIFO_POP_DELAY	1 /* 1 us*/
 u32 hl_rst_src_get(void)
 {
 	u32 fifo_sts, fill_level;
@@ -237,9 +238,12 @@
 	}
 
 	fill_level = fifo_sts >> PSOC_BOOT_CONF_BTM_RST_FIFO_STS_FILL_LEVEL_S;
-	for (; fill_level > 1; fill_level--)
+
+	for (; fill_level > 1; fill_level--) {
 		regmap_writel(PSOC_BOOT_CONF_BTM_RST_FIFO_CTRL_POP_M,
 					D0_PSOC_BOOT_CONF_REG(mmPSOC_BOOT_CONF_BTM_RST_FIFO_CTRL));
+		udelay(HL_BTM_FIFO_POP_DELAY);
+	}
 
 	rst_src.src = regmap_readl(D0_PSOC_BOOT_CONF_REG(mmPSOC_BOOT_CONF_BTM_RST_FIFO_HEAD)) &
 					BTM_FSM_FIFO_HEAD_RST_SRC_MASK;
